src/tests/integration/test_generic_db_operations.py

```diff
@@ -1253,3 +1253,100 @@ def test_autocommit_create_table_multiple_cells(
     ).result
 
     assert len(result) == 3
+
+
+@pytest.mark.parametrize(
+    "ip_with_dynamic_db, snippet_name, error_msgs, error_type",
+    [
+        (
+            "ip_with_postgreSQL",
+            "mysnippet",
+            [
+                "function not_a_function(integer) does not exist",
+                "No function matches the given name and argument types",
+            ],
+            "RuntimeError",
+        ),
+        (
+            "ip_with_postgreSQL",
+            "mysnip",
+            [
+                "If using snippets, you may pass the --with argument explicitly.",
+                'relation "mysnip" does not exist',
+            ],
+            "RuntimeError",
+        ),
+        (
+            "ip_with_mySQL",
+            "mysnippet",
+            [
+                "FUNCTION db.not_a_function does not exist",
+            ],
+            "RuntimeError",
+        ),
+        (
+            "ip_with_mySQL",
+            "mysnip",
+            [
+                "If using snippets, you may pass the --with argument explicitly.",
+                "Table 'db.mysnip' doesn't exist",
+            ],
+            "RuntimeError",
+        ),
+        (
+            "ip_with_mariaDB",
+            "mysnippet",
+            [
+                "FUNCTION db.not_a_function does not exist",
+            ],
+            "RuntimeError",
+        ),
+        (
+            "ip_with_mariaDB",
+            "mysnip",
+            [
+                "If using snippets, you may pass the --with argument explicitly.",
+                "Table 'db.mysnip' doesn't exist",
+            ],
+            "RuntimeError",
+        ),
+    ],
+    ids=[
+        "no-typo-postgreSQL",
+        "with-typo-postgreSQL",
+        "no-typo-mySQL",
+        "with-typo-mySQL",
+        "no-typo-mariaDB",
+        "with-typo-mariaDB",
+    ],
+)
+def test_query_snippet_invalid_function_error_message(
+    request, ip_with_dynamic_db, snippet_name, error_msgs, error_type
+):
+    # Set up snippet
+    ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+    ip_with_dynamic_db.run_cell(
+        """
+                %sql CREATE TABLE IF NOT EXISTS penguins (id INTEGER)
+                %sql INSERT INTO penguins VALUES (1)
+                """
+    )
+    ip_with_dynamic_db.run_cell(
+        """
+            %%sql --save mysnippet
+            SELECT * FROM penguins
+            """
```

Comment:
What's the issue with the other DBs?

Reply:
I realized that the issues may have been local so I am running integration tests on the CI to get a better idea. Will have an update soon

Reply:
Was able to work through most of the DB issues - leaving only one xfail for [MSSQL](https://github.com/ploomber/jupysql/pull/912/files#diff-5bfb2487a0c01ba83dbc1d70b2fa3bec84d0d20ab5000c76379915572bd176e6R1321-R1331) and two for [duckDB_native](https://github.com/ploomber/jupysql/pull/912/files#diff-5bfb2487a0c01ba83dbc1d70b2fa3bec84d0d20ab5000c76379915572bd176e6R1396-R1411). Ready for review @neelasha23 @nooodle-soup

Reply:
Maybe duckdb is throwing some other exception and not catalog error? Please open issues for these two cases so that it can be tackled later.